### PR TITLE
Add feature flags to config, implement for messages

### PR DESCRIPTION
**Pull request by luoser**
## what
1. Added `features` to our config model so we can start tapping into our feature flags to use them! Config is already a part of our `environment` so this was a breeze.

2. Implemented flag for messages in our `ProfileViewModel`.

3. Modernized `ProfileViewModelTest` and added additional tests.

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.java [hunk 1]
```diff
@@ -33,19 +34,22 @@
...
   private ProfileAdapter adapter;
   private RecyclerViewPaginator paginator;
 
   protected @Bind(R.id.avatar_image_view) ImageView avatarImageView;
-  protected @Bind(R.id.user_name_text_view) TextView userNameTextView;
-  protected @Bind(R.id.created_count_text_view) TextView createdCountTextView;
   protected @Bind(R.id.backed_count_text_view) TextView backedCountTextView;
-  protected @Bind(R.id.created_text_view) TextView createdTextView;
   protected @Bind(R.id.backed_text_view) TextView backedTextView;
+  protected @Bind(R.id.created_count_text_view) TextView createdCountTextView;
```
Comment (luoser) [inline app/src/main/java/com/kickstarter/ui/activities/ProfileActivity.java]: lotta noise here from alphabetizing

File: app/src/main/java/com/kickstarter/libs/CurrentUser.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ public void refresh(final @NonNull User freshUser) {
   }
 
   @Override
-  public Observable<User> observable() {
+  public @NonNull Observable<User> observable() {
```
Comment (luoser) [inline app/src/main/java/com/kickstarter/libs/CurrentUser.java]: our [interface's annotation of this](https://github.com/kickstarter/android-oss/blob/master/app/src/main/java/com/kickstarter/libs/CurrentUserType.java#L38) was also `@NonNull` so we should be consistent

File: app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java [hunk 3]
```diff
@@ -141,6 +148,9 @@ public ViewModel(final @NonNull Environment environment) {
       )
         .map(p -> p.first || p.second);
 
+      this.messagesButtonHidden = this.currentConfig.observable()
+        .map(config -> !config.features().getOrDefault("android_messages", false));
```
Comment (mbrandonw) [inline app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java]: so I think we want the default to be `true`, i.e. if we deleted the feature flag from the server we would interpret that as the feature being _on_ for everyone, not _off_.

does that make sense?

File: app/src/test/java/com/kickstarter/viewmodels/ProfileViewModelTest.java [hunk 4]
```diff
@@ -210,38 +178,49 @@ public void testProfileViewModel_EmitsUserNameAndAvatar() {
...
+
+    this.avatarImageViewUrl.assertValues(user.avatar().medium());
+    this.userNameTextViewText.assertValues(user.name());
+  }
+
+  @Test
+  public void testProfileViewModel_MessagesButtonHidden() {
+    final Config config = ConfigFactory.config()
+      .toBuilder()
+      .features(Collections.singletonMap("android_messages", false))
```
Comment (mbrandonw) [inline app/src/test/java/com/kickstarter/viewmodels/ProfileViewModelTest.java]: what about the case when `"android_messages": true` _and_ `"android_messages"` is absent? 🤔

File: app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java (hunk 3 above)
Comment (luoser) [inline app/src/main/java/com/kickstarter/viewmodels/ProfileViewModel.java]: bitmoji

File: app/src/test/java/com/kickstarter/viewmodels/ProfileViewModelTest.java (hunk 4 above)
Comment (luoser) [inline app/src/test/java/com/kickstarter/viewmodels/ProfileViewModelTest.java]: 🤔 

i'll add more tests

**Pull request merged**